scripts/ci/upload-r2-object.py

```
@@ -0,0 +1,138 @@
+                    "method": request.get_method(),
+                    "url": request.full_url,
+                    "headers": dict(request.header_items()),
+                    "body_sha256": hashlib.sha256(body).hexdigest(),
+                },
+                sort_keys=True,
+            )
+        )
+        return 0
+
+    try:
+        with urllib.request.urlopen(request, timeout=30) as response:
```

Ruff (0.15.17)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)